Userland/Libraries/LibWeb/Layout/Box.cpp

```diff
@@ -56,7 +56,10 @@ bool Box::is_scroll_container() const
 
 bool Box::is_user_scrollable() const
 {
-    // FIXME: Support horizontal scroll as well (overflow-x)
+    // FIXME: Support horizontal scroll as well (overflow-x).
+	if (computed_values().overflow_y() == CSS::Overflow::Hidden || computed_values().overflow_x() == CSS::Overflow::Hidden) 
+		return false;
```

Comment:
ignoring scrolling if either of directions has "overflow: hidden" is wrong: vertical scrolling should not be ignored, if only horizontal direction has hidden overflow.

Reply:
assuming you trying to fix:
```html
<!doctype html><style>
* { font-size: 200px; }
main { overflow-x: hidden; }
</style>
<main>
1<br>
2<br>
3<br>
4<br>
5<br>
6<br>
7<br>
8<br>
9<br>
```

let's have a look paintable tree dump:
```
ViewportPaintable (Viewport<#document>) [0,0 1279x780] overflow: [0,0 1279x2086]
  PaintableWithLines (BlockContainer<HTML>) [0,0 1279x2086]
    PaintableWithLines (BlockContainer<BODY>) [8,8 1263x2070]
      PaintableWithLines (BlockContainer<MAIN>) [8,8 1263x2070]
        TextPaintable (TextNode<#text>)
        TextPaintable (TextNode<#text>)
        TextPaintable (TextNode<#text>)
        TextPaintable (TextNode<#text>)
        TextPaintable (TextNode<#text>)
        TextPaintable (TextNode<#text>)
        TextPaintable (TextNode<#text>)
        TextPaintable (TextNode<#text>)
        TextPaintable (TextNode<#text>)
```

the only paintable with scrollable overflow (named overflow in the dump) is viewport. `<main>` does not have any. so `<main>` is scrollable based on css overflow properties, because overflow_y is auto.
BUT it does not have any scrollable overflow. let's make `is_user_scrollable()` to also look at rect returned by `PaintableBox::scrollable_overflow_rect()`?